src/middlewared/middlewared/service.py

```diff
@@ -161,14 +161,21 @@ def __init__(self, middleware):
 
 class ServiceChangeMixin:
     async def _service_change(self, service, verb):
-        enabled = (await self.middleware.call(
-            'datastore.query', 'services.services', [('srv_service', '=', service)], {'get': True}
-        ))['srv_enable']
 
-        started = await self.middleware.call(f'service.{verb}', service, {'onetime': False})
+        svc_obj = await self.middleware.call(
+            'service.query',
+            [('service', '=', service)],
+            {'get': True}
+        )
+
+        if svc_obj['state'].lower() == 'running':
+            started = await self.middleware.call(f'service.{verb}', service, {'onetime': True})
 
-        if enabled and not started:
-            raise CallError(f'The {service} service failed to start', CallError.ESERVICESTARTFAILURE)
+            if svc_obj['enable'] and not started:
```

Comment:
I think `svc_obj['enable']` can be removed. If the service is not set to start at boot and was running, it should not have failed to start.